semanticcpg/src/main/scala/io/shiftleft/semanticcpg/utils/ExpandTo.scala

```diff
@@ -17,91 +17,74 @@ object ExpandTo {
   // For languages which make use of function pointers, this can also be the
   // pointer itself.
   // TODO move into traversal dsl - used in codescience
-  def callReceiverOption(callNode: Vertex): Option[nodes.StoredNode] = {
-    callNode
-      .asInstanceOf[nodes.StoredNode]
-      ._receiverOut
-      .nextOption
-  }
+  def callReceiverOption(callNode: nodes.Call): Option[nodes.StoredNode] =
+    callNode.receiverOut.nextOption
 
-  def callReceiver(callNode: Vertex): nodes.StoredNode = callReceiverOption(callNode).get
+  def callReceiver(callNode: nodes.Call): nodes.StoredNode =
+    callReceiverOption(callNode).get
 
   // TODO move into traversal dsl - used in codescience
-  def callArguments(callNode: Vertex): Iterator[nodes.Expression] = {
-    callNode.asInstanceOf[nodes.StoredNode]._argumentOut.asScala.map(_.asInstanceOf[nodes.Expression])
-  }
+  def callArguments(callNode: nodes.CallRepr): Iterator[nodes.Expression] =
+    callNode._argumentOut.asScala.map(_.asInstanceOf[nodes.Expression])
```

Comment:
Why are we sometimes using `_argumentOut` and sometimes `argumentOut`?
We now see to have all of the xxxIN/OUT method with and without underliner. Whats the reasoning?

Reply:
The difference between them is that `_xxxIn|Out` are defined on `StoredNode`, which means that every node has in/out accessors for *every* edge, even if that's not a legal edge at that position. 
Since we statically know which nodes allow which edges (and also know the neighbor node type) I wanted to replace those with more specific accessors, i.e. define them on the node type (rather than StoredNode), and narrow the type as much as possible. Example: 
```scala
class MethodReturn {
  override def _astIn(): JIterator[StoredNode] = ...
  def astIn: JIterator[Method] = ...
}
```

We had a discussion between us and @bbrehm on slack about this, and came to the conclusion that while it makes sense to add the more specific accessors (more type safety, less casts, help in IDE), @bbrehm pointed out there are still some places (like this one) where we currently don't get around using the 'generic' `_xxxIn|Out` accessors. 
So I'm using the specific accessors wherever possible for now.